### PR TITLE
8269827: JMH tests for AES/GCM byte[] and bytebuffers

### PR DESCRIPTION
**Pull request by ascarpino**
Hi,

I need a review of these new jmh tests that run AES/GCM encryption and decryption using byte[], heap bytebuffers, and direct bytebuffers as input and output buffers for single and multi-part testing. 

thanks

Tony

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269827](https://bugs.openjdk.java.net/browse/JDK-8269827): JMH tests for AES/GCM byte[] and bytebuffers

### Reviewers
 * [Eric Caspole](https://openjdk.java.net/census#ecaspole) (@ericcaspole - Committer) ⚠️ Review applies to 0c7113ef45c2281e1d65abf6468474749d24055c
 * [Weijun Wang](https://openjdk.java.net/census#weijun) (@wangweij - **Reviewer**) ⚠️ Review applies to 2a7d1df664c77c66781f3de31ccbd24c1684f209

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4672/head:pull/4672` \
`$ git checkout pull/4672`

Update a local copy of the PR: \
`$ git checkout pull/4672` \
`$ git pull https://git.openjdk.java.net/jdk pull/4672/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4672`

View PR using the GUI difftool: \
`$ git pr show -t 4672`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4672.diff">https://git.openjdk.java.net/jdk/pull/4672.diff</a>

</details>

### CODE REVIEW

File: test/micro/org/openjdk/bench/javax/crypto/full/AESGCMBench.java [hunk 1]
```diff
@@ -26,41 +26,38 @@
...
 public class AESGCMBench extends CryptoBase {
 
-    @Param({"AES/GCM/NoPadding"})
-    private String algorithm;
-
     @Param({"128"})
     private int keyLength;
 
-    @Param({""+16*1024})
+    @Param({"" + 1024, "" +  1500, "" + 4096, "" + 16384})
```
Comment (wangweij) [inline test/micro/org/openjdk/bench/javax/crypto/full/AESGCMBench.java:46]: Now that you don't calculate anymore, how about just string literals like `"1024"` etc?

File: test/micro/org/openjdk/bench/javax/crypto/full/AESGCMByteBuffer.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+/**
+ * This performance tests runs AES/GCM encryption and decryption using heap and
+ * direct ByteBuffers as input and output buffers for single and multi-part
+ * operations.
+ *
+ * This test rotates the IV and creates a new GCMParameterSpec for each encrypt
+ * benchmark operation
+ */
+
```
Comment (wangweij) [inline test/micro/org/openjdk/bench/javax/crypto/full/AESGCMByteBuffer.java:44]: Do you want to create a "small" version?

File: test/micro/org/openjdk/bench/javax/crypto/full/AESGCMBench.java [hunk 3]
```diff
@@ -26,41 +26,38 @@
...
     private int dataSize;
 
-    byte[] data;
     byte[] encryptedData;
+    byte[] in, out;
     private Cipher encryptCipher;
     private Cipher decryptCipher;
     SecretKeySpec ks;
     GCMParameterSpec gcm_spec;
-    byte[] aad;
```
Comment (wangweij) [inline test/micro/org/openjdk/bench/javax/crypto/full/AESGCMBench.java:58]: Not an expert on this, but why no more AAD?

Comment (ascarpino) [reply test/micro/org/openjdk/bench/javax/crypto/full/AESGCMBench.java:58]: For a performance test there isn't any value in adding an AAD.  It's data sitting in a buffer that an be easily replicated with a multipart operation

File: test/micro/org/openjdk/bench/javax/crypto/full/AESGCMBench.java (hunk 1 above)
Comment (ascarpino) [reply test/micro/org/openjdk/bench/javax/crypto/full/AESGCMBench.java:46]: Funny.. I didn't even notice that

File: test/micro/org/openjdk/bench/javax/crypto/full/AESGCMByteBuffer.java (hunk 2 above)
Comment (ascarpino) [reply test/micro/org/openjdk/bench/javax/crypto/full/AESGCMByteBuffer.java:44]: I debated it.. I'm still not sure, but I guess I could.

**Comment by ericcaspole** [comment]
From the JMH point of view, looks good to me.

**Review by ericcaspole — APPROVED** [review]
From the JMH point of view, looks good to me.

File: test/micro/org/openjdk/bench/javax/crypto/full/AESGCMByteBuffer.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+public class AESGCMByteBuffer extends CryptoBase {
+
+    @Param({"128"})
+    private int keyLength;
+
+    @Param({"" + 1024, "" + 1500, "" + 4096, "" + 16384})
```
Comment (wangweij) [inline test/micro/org/openjdk/bench/javax/crypto/full/AESGCMByteBuffer.java:50]: You can update the line above as well.

**Review by wangweij — APPROVED** [review]

File: test/micro/org/openjdk/bench/javax/crypto/full/AESGCMByteBuffer.java (hunk 4 above)
Comment (ascarpino) [reply test/micro/org/openjdk/bench/javax/crypto/full/AESGCMByteBuffer.java:50]: Grr.. I'm not sure how I missed that one

**Pull request closed without merging**